app/src/main/java/me/ccrama/redditslide/OpenRedditLink.java

```diff
@@ -15,58 +16,49 @@
     private final static String TAG = "OpenRedditLink";
 
     public OpenRedditLink(Context context, String url) {
-        boolean np = url.contains("np.");
+        boolean np = url.matches("(https?://)?np.reddit.com.*");
+
         url = url.replace("np.", "");
```

Comment:
Worth removing all the language subdomains too rather than just `np.`? regex would be `([_A-Za-z0-9]{2}-)?[_A-Za-z0-9]{2}\\.`. eg en-us.reddit.com, de.reddit.com, though technically anything even __-__.reddit.com is a working URL